Libraries/Opc.Ua.Server/Server/ReverseConnectServer.cs

```diff
@@ -171,26 +174,30 @@ private void OnReverseConnect(object state)
                 foreach (var reverseConnection in m_connections.Values)
                 {
                     // recharge a rejected connection after timeout
-                    if (reverseConnection.State == ReverseConnectState.Rejected &&
+                    if (reverseConnection.LastState == ReverseConnectState.Rejected &&
                         reverseConnection.RejectTime + TimeSpan.FromMilliseconds(m_rejectTimeout) < DateTime.UtcNow)
                     {
-                        reverseConnection.State = ReverseConnectState.Closed;
+                        reverseConnection.LastState = ReverseConnectState.Closed;
                     }
+
                     // try the reverse connect
-                    if (reverseConnection.State == ReverseConnectState.Closed)
+                    if (reverseConnection.MaxSessionCount == 0 ||
+                       (reverseConnection.MaxSessionCount == 1 &&
+                        reverseConnection.LastState == ReverseConnectState.Closed) ||
+                        reverseConnection.MaxSessionCount > ServerInternal.SessionManager.GetSessions().Count)
```

Comment:
the session count is not exactly right here the number of reverse connect sessoin but at least it provides some upper boundary when to stop pinging.